tests/integration2/tunnel/tunnel_test.go

```diff
@@ -0,0 +1,351 @@
+spec:
+  host: {{ .ingressDNS }}
+  exportTo: [ "." ]
+  subsets:
+  - name: client-2
+    trafficPolicy:
+      portLevelSettings:
+      - port:
+          number: 443
+        tls:
+          caCertificates: /etc/istio/tunnel-certs/ca.crt
+          clientCertificate: /etc/istio/tunnel-certs/client.crt
+          mode: MUTUAL
+          privateKey: /etc/istio/tunnel-certs/client.key
+          sni: {{ .ingressDNS }}
+          subjectAltNames:
+          - {{ .ingressDNS }}
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  creationTimestamp: null
+  name: ingress-service-entry
+spec:
+  endpoints:
+  - address: {{.ingressAddress}}
+  hosts:
+  - {{ .ingressDNS }}
+  ports:
+  - name: index-binding-id-7777
+    number: {{.ingressPort}}
+    protocol: TCP
+  resolution: STATIC
+`
+
+	clientSideConfig = `
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: mesh-to-egress-client
+spec:
+  gateways:
+  - mesh
+  hosts:
+  - {{ .serviceName }}
+  tcp:
+  - match:
+    - destinationSubnets:
+      - {{ .vip }}
+      gateways:
+      - mesh
+    route:
+    - destination:
+        host: istio-egressgateway.{{ .systemNamespace }}.svc.cluster.local
+        port:
+          number: 443
+        subset: client
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: sidecar-to-egress-client
+spec:
+  host: istio-egressgateway.{{ .systemNamespace }}.svc.cluster.local
+  subsets:
+  - name: client
+    trafficPolicy:
+      tls:
+        mode: ISTIO_MUTUAL
+        sni: {{ .sidecarSNI }}
+`
+	serverSideConfig = `---
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  creationTimestamp: null
+  name: index-binding-id-gateway-tls
+spec:
+  selector:
+    istio: ingressgateway # use Istio default gateway implementation
+  servers:
+  - hosts:
+    - "*"
+    port:
+      name: tls
+      number: 443
+      protocol: TLS
+    tls:
+      caCertificates: /etc/istio/tunnel-certs/ca.crt
+      mode: MUTUAL
+      privateKey: /etc/istio/tunnel-certs/service.key
+      serverCertificate: /etc/istio/tunnel-certs/service.crt
+      subjectAltNames:
+      - {{ .clientSAN }}
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  creationTimestamp: null
+  name: index-binding-id-virtual-service-tls
+spec:
+  gateways:
+  - index-binding-id-gateway-tls
+  hosts:
+  - {{ .ingressDNS }}
+  tcp:
+  - route:
+    - destination:
+        host: index-binding-id.service-fabrik
+        port:
+          number: {{ .port }}
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  creationTimestamp: null
+  name: index-binding-id-service-entry
+spec:
+  endpoints:
+  - address: {{ .address }}
+  hosts:
+  - index-binding-id.service-fabrik
+  ports:
+  - name: index-binding-id-7777
+    number: {{ .port }}
+    protocol: TCP
+  resolution: STATIC
+
+`
+)
+
+func TestTunnel(t *testing.T) {
```

Comment:
It would be good to add a description of what this test is doing. Something to the effect of tunneling TCP communication between service a and b, such that the traffic is routed a->egress->ingress->b.

Reply:
We will add a description